### PR TITLE
proc: add FindProcessByCmdline

### PR DESCRIPTION
**Pull request by derekbit**
Longhorn/longhorn#7263

### CODE REVIEW

File: proc/proc.go [hunk 1]
```diff
@@ -175,3 +176,48 @@ func FindProcessByName(name string) (*os.Process, error) {
...
+	if err != nil {
+		return nil, err
+	}
+
+	for _, p := range ps {
+		if p.IsDir() {
+			pid, convErr := strconv.Atoi(p.Name())
+			if convErr != nil {
+				continue
+			}
```
Comment (innobead) [inline proc/proc.go:200]: Have you thought about reusing the code from `go-ps` lib? So don't need to do parsing by yourself. 

```go
	processes, err := ps.Processes()
	if err != nil {
		return nil, fmt.Errorf("failed to list processes")
	}

	for _, process := range processes {
        // TODO your implementation ..
	}
```

**Review by innobead — CHANGES_REQUESTED** [review]
LGTM, but one comment only needs to resolve.

File: proc/proc.go (hunk 1 above)
Comment (derekbit) [reply proc/proc.go:200]: Yes, I thought the idea, but go-ps gets process name from `/proc/<pid>/status` rather than `/proc/<pid>/cmdline`.

Comment (innobead) [reply proc/proc.go:200]: I see, but what I meant is you can use the lib to do below instead, so you don't need to scan the proc directory again to get all processes, just reuse the lib. WDYT? 

```go
func FindProcessByCmdline(cmdline string) ([]*os.Process, error) {
	processes, err := ps.Processes()
	if err != nil {
		return nil, fmt.Errorf("failed to list processes")
	}

	var foundProcesses []*os.Process
	
	for _, process := range processes {
		cmdlinePath := filepath.Join("/proc", strconv.Itoa(process.Pid()), "cmdline")
		cmdlineContent, readErr := os.ReadFile(cmdlinePath)
		if readErr != nil {
			continue
		}

		if strings.HasPrefix(string(cmdlineContent), cmdline) {
			process, err := os.FindProcess(process.Pid())
			if err != nil {
				return nil, err
			}
			foundProcesses = append(foundProcesses, process)
		}
	}
	
	if len(foundProcesses) == 0 {
		return nil, fmt.Errorf("process with cmdline %s is not found", cmdline)
	}

	return foundProcesses, nil
}
```

Comment (derekbit) [reply proc/proc.go]: Ah, misunderstood. Addressed, thank you.

**Pull request merged**